src/EditorFeatures/CSharp/TextStructureNavigation/TextStructureNavigatorProvider.cs

```diff
@@ -34,9 +35,14 @@ protected override bool IsWithinNaturalLanguage(SyntaxToken token, int position)
             switch (token.Kind())
             {
                 case SyntaxKind.StringLiteralToken:
-                    // Before the " is considered outside the string
-                    // TODO: does this handle verbatim strings correctly?
-                    return position > token.SpanStart;
+                    // This, in combination with the override of GetExtentOfWordFromToken() below, treats the closing
+                    // quote as a separate token.  This maintains behavior with VS2013.
+                    if (position == token.Span.End - 1 && token.Text.EndsWith("\"", StringComparison.Ordinal))
```

Comment:
I tried changing this to a single return statement but since the condition needed to be inverted it was difficult to read/understand, so for readability's sake I'm going to leave this.